sources/ElkArte/EventManager.php

```diff
@@ -235,47 +237,34 @@ protected function _register_events($classes)
 	}
 
 	/**
-	 * Registers an event at a certain position with a defined priority.
+	 * Reflects a specific class method to see what parameters are needed
 	 *
-	 * @param string $position The position at which the event will be triggered
-	 * @param mixed[] $event An array describing the event we want to trigger:
-	 *   0 => string - the position at which the event will be triggered
-	 *   1 => string[] - the class and method we want to call:
-	 *      array(
-	 *        0 => string - name of the class to instantiate
-	 *        1 => string - name of the method to call
-	 *      )
-	 *   2 => null|string[] - an array of dependencies in the form of strings representing the
-	 *        name of the variables the method requires.
-	 *        The variables can be from:
-	 *          - the default list of variables passed to the trigger
-	 *          - properties (private, protected, or public) of the object that instantiate the \ElkArte\EventManager
-	 *            (i.e. the controller)
-	 *          - globals
-	 * @param int $priority Defines the order the method is called.
+	 * Currently only checks on number required, can be expanded to make use of
+	 * $params = $r->getParameters() and then $param-> getName isOptional etc
+	 * to ensure required named are being passed.
+	 *
+	 * @param string $class_name
+	 * @param string $method_name
+	 * @param array $dependencies the dependencies the event registered
 	 */
-	public function register($position, $event, $priority = 0)
+	protected function _checkParameters($class_name, $method_name, &$dependencies)
 	{
-		if (!isset($this->_registered_events[$position]))
+		// Lets check on the actual methods parameters
+		try
 		{
-			$this->_registered_events[$position] = new Event(new Priority());
+			$r = new \ReflectionMethod($class_name, $method_name);
+			$number_params = $r->getNumberOfParameters();
+			unset($r);
 		}
-
-		$this->_registered_events[$position]->add($event, $priority);
-	}
-
-	/**
-	 * Gets the names of all the classes already loaded.
-	 *
-	 * @return string[]
-	 */
-	protected function _declared_classes()
-	{
-		if ($this->_declared_classes === null)
+		catch (\Exception $e)
 		{
-			$this->_declared_classes = get_declared_classes();
+			$number_params = 0;
 		}
 
-		return $this->_declared_classes;
+		// Php8 will not like passing parameters to a method that takes none
```

Comment:
php8 is annoying. :laughing:

Reply:
Its VERY particular about parameters now, number passed, ordering,  optional ones need to be at the end, etc.

Reply:
Hummm...looks like a git merge/munge error ... Guess it thinks public function register() is not important anymore DOH!